src/geometry/web_mercator_rect.rs

```diff
@@ -0,0 +1,140 @@
+//! A Web Mercator axis-aligned rectangle.
+
+use crate::math::base::{HasAabbIntersector, PointCulling};
+use crate::math::sat::{CachedAxesIntersector, ConvexPolyhedron, Intersector};
+use crate::math::web_mercator::WebMercatorCoord;
+use arrayvec::ArrayVec;
+use nalgebra::{Point3, RealField, Unit, Vector2};
+use nav_types::{ECEF, WGS84};
+use serde::{Deserialize, Serialize};
+
+/// The dead sea is at -413m, but we use a more generous minimum
+const MIN_ELEVATION_M: f64 = -1000.0;
+/// Mt. Everest is at 8,848m, plus we need some safety margin
+const MAX_ELEVATION_M: f64 = 9000.0;
+
+/// A rectangle on a Web Mercator map, not rotated.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WebMercatorRect {
+    north_west: WebMercatorCoord,
+    south_east: WebMercatorCoord,
+}
+
+impl WebMercatorRect {
+    /// Returns `None` when `z` is greater than [`MAX_ZOOM`](index.html#constant.max_zoom)
+    /// or when the coordinates are out of bounds for the zoom level `z`. It also returns `None`
+    /// when the rect is larger than
+    /// Rects crossing the ±180° longitude line are not supported at the moment.
+    pub fn new(min: Vector2<f64>, max: Vector2<f64>, z: u8) -> Option<Self> {
+        let north_west = WebMercatorCoord::from_zoomed_coordinate(nalgebra::inf(&min, &max), z)?;
+        let south_east = WebMercatorCoord::from_zoomed_coordinate(nalgebra::sup(&min, &max), z)?;
```

Comment:
Nit: Use `min.inf(&max)` and `min.sup(&max)` instead, as the `nalgebra::...` versions are deprecated?

Reply:
The inherent methods are not there yet in version 0.20.0. We can do that here and in Aabb when we switch to 0.21.0

Reply:
Ok, can you please leave a `TODO` here?

Reply:
It's already there.